src/App.tsx

```diff
@@ -1,223 +1,199 @@
+      event.preventDefault();
+      setError(ErrorNotification.None);
+
+      if (title.trim() && user) {
+        setIsAdding(true);
+
+        try {
+          await addTodo({
+            userId: user.id,
+            title: title.trim(),
+            completed: false,
+          });
+
+          await loadingTodos();
+
+          setTitle('');
+        } catch {
+          setError(ErrorNotification.Add);
+        } finally {
+          setIsAdding(false);
+        }
+      } else {
+        setError(ErrorNotification.Title);
+      }
+    }, [title, user],
+  );
+
+  const handleRemove = useCallback(
+    async (todoId: number) => {
+      setError(ErrorNotification.None);
+      setLoadTodosIds(prevIds => [...prevIds, todoId]);
+
+      try {
+        await removeTodo(todoId);
+
+        await loadingTodos();
+      } catch {
+        setError(ErrorNotification.Remove);
+      } finally {
+        setLoadTodosIds([]);
+      }
+    }, [completedTodos],
+  );
+
+  const removeCompletedTodos = useCallback(
+    async () => {
+      setError(ErrorNotification.None);
+      setLoadTodosIds(prevTodoIds => ([
+        ...prevTodoIds,
+        ...completedTodos.map(todo => todo.id),
+      ]));
+
+      try {
+        await Promise.all(completedTodos.map(todo => (
+          removeTodo(todo.id)
+        )));
+
+        await loadingTodos();
+      } catch {
+        setError(ErrorNotification.Remove);
+      } finally {
+        setLoadTodosIds([]);
+      }
+    }, [completedTodos],
+  );
+
+  const visibleTodos = useMemo(() => (
+    todos.filter(todo => {
+      switch (filter) {
+        case FilterType.Completed:
+          return todo.completed;
+
+        case FilterType.Active:
+          return !todo.completed;
+
+        case FilterType.All:
+        default:
+          return todo;
+      }
+    })
+  ), [todos, filter]);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
 
       <div className="todoapp__content">
         <header className="todoapp__header">
-          <button
-            data-cy="ToggleAllButton"
-            type="button"
-            className="todoapp__toggle-all active"
-          />
-
-          <form>
-            <input
-              data-cy="NewTodoField"
-              type="text"
-              ref={newTodoField}
-              className="todoapp__new-todo"
-              placeholder="What needs to be done?"
-            />
-          </form>
-        </header>
-
-        <section className="todoapp__main" data-cy="TodoList">
-          <div data-cy="Todo" className="todo completed">
-            <label className="todo__status-label">
-              <input
-                data-cy="TodoStatus"
-                type="checkbox"
-                className="todo__status"
-                defaultChecked
-              />
-            </label>
-
-            <span data-cy="TodoTitle" className="todo__title">HTML</span>
+          {todos.length > 0 && (
```

Comment:
```suggestion
          {todos && (
```

Reply:
I disagree. JS Style guide says otherwise: https://mate-academy.github.io/style-guides/javascript.html#153use-shortcuts-for-booleans-but-explicit-comparisons-for-strings-and-numbers